### ISSUE TITLE
expect_error fails for java Exceptions

### ISSUE THREAD

**Issue by pfistfl** [r-lib/testthat#964]
since version 2.3.0, tests that check for java Exceptions fail. 

**On 2.2.0**: 
`remotes::install_version("testthat", "2.2.0")`
``` r
library(rJava)
library(testthat)
.jinit()
.jnew("hello/world", check = TRUE)
#> Error in .jnew("hello/world", check = TRUE): java.lang.ClassNotFoundException

expect_error(.jnew("hello/world", check = TRUE))
```
<sup>Created on 2019-11-08 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**On 2.3.0**: 
`remotes::install_version("testthat", "2.3.0")`
``` r
library(rJava)
library(testthat)
.jinit()
.jnew("hello/world", check = TRUE)
#> Error in .jnew("hello/world", check = TRUE): java.lang.ClassNotFoundException

expect_error(.jnew("hello/world", check = TRUE))
#> Error in ._jobjRef_dollar(x[["jobj"]], name): no field, method or inner class called 'trace'
```
<sup>Created on 2019-11-08 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

I digged a little deeper, and the reason is most likely that `rlang::cnd_entrace` used in `capture_error` tries to access the `$trace` slot, which does prompt a java Error.

Is there another method intended to test for Exceptions?

**Comment by hadley** [comment]
This is a known bug and we're hoping to get a fix out next week.

**Comment by lionel-** [comment]
Fixed by #966.
